storage: fix reason in "initiating a merge" message

tbg opened the pull request:
It was alluding to a "threshold" but wasn't actually printing it. Now it
does. This will be helpful to understand why merges happened.

Release note: None

tbg committed:
storage: fix reason in "initiating a merge" message

It was alluding to a "threshold" but wasn't actually printing it. Now it
does. This will be helpful to understand why merges happened.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/44282)
<!-- Reviewable:end -->

tbg committed:
server: add comment about decoy Health endpoint

We're not really using this one, but this is the only one one will
reasonably discover in the code. Leave a helpful comment pointing
folks to the real one.

Release note: None

tbg commented:
bors r=knz
